From ben-linux@fluff.org  Tue Jun  1 13:40:21 2010
From: Maurus Cuelenaere <mcuelenaere@gmail.com>
Date: Tue, 25 May 2010 05:36:49 +0100
Subject: USB: s3c_hsotg: define USB_GADGET_DUALSPEED in Kconfig
To: linux-samsung-soc@vger.kernel.org, linux-usb@vger.kernel.org
Cc: Maurus Cuelenaere <mcuelenaere@gmail.com>, Ben Dooks <ben-linux@fluff.org>
Message-ID: <1274762213-3404-3-git-send-email-ben-linux@fluff.org>


From: Maurus Cuelenaere <mcuelenaere@gmail.com>

The s3c_hsotg driver sets usb_gadget->is_dualspeed to 1, yet it doesn't define
USB_GADGET_DUALSPEED in Kconfig. This triggers a NULL pointer dereference in
the composite driver (which is fixed in another patch).

Signed-off-by: Maurus Cuelenaere <mcuelenaere@gmail.com>
Signed-off-by: Ben Dooks <ben-linux@fluff.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/gadget/Kconfig
+++ b/drivers/usb/gadget/Kconfig
@@ -295,6 +295,7 @@ config USB_GADGET_S3C_HSOTG
 	boolean "S3C HS/OtG USB Device controller"
 	depends on S3C_DEV_USB_HSOTG
 	select USB_GADGET_S3C_HSOTG_PIO
+	select USB_GADGET_DUALSPEED
 	help
 	  The Samsung S3C64XX USB2.0 high-speed gadget controller
 	  integrated into the S3C64XX series SoC.
